### ISSUE TITLE
[Accessibility]: Notebooks read-only icon color contrast

### ISSUE THREAD

**Issue by marisak** [sourcegraph/sourcegraph#35684]
### Audit type

ARC Toolkit

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/35142

### Problem description

There are a lot of failures of the ARC test `insufficientNormalTextContrast` while reading a notebook. When using the hover in Chrome Inspector, it looks like most of the color contrast passes WCAG 2.1 AA requirements (greater than 4.5). One part of the page that could use reexamination is the `#` anchors on the headers; however strictly speaking these aren't meant to be semantic text but anchors, so that's a separate problem. 

### Expected behavior

- Higher color contrast ratios for some of this text 
- Decorative elements are distinguished properly 

### Additional details

This might overlap significantly with other Notebooks issues but I wasn't able to find them. 

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.

**Comment by limitedmage** [comment]
I no longer see any color contrast errors on the Notebook page, changes may have been made that fixed it (eg. new syntax highlighting theme).
